app/models/option_node.rb

```diff
@@ -242,6 +242,8 @@ def arrange_as_json(hash = nil)
         # And option_has_answers? kicks off a big SQL query for a huge set.
         # Conditions association should be eager loaded.
         unless huge?
+          # This should ideally consult the `destroy` permission for option_node in the authorization system,
+          # but it all needs to be refactored anyway; it's not worth doing more gross things just for this.
           branch[:removable] = !option_set.option_has_answers?(node.option_id) && node.conditions.empty?
```

Comment:
Is this comment still relevant? You asked me to document this while pairing but I'm not completely sure what it's referring to.